### DISCUSSION TITLE
.C2 for CPP2 source extension?

### DISCUSSION THREAD

**Discussion by jamadagni** [hsutter/cppfront#1241] [Suggestions]
I noticed that we have `.h2` for CPP2 header extension. If so `.c2` is the simplest source extension no? Why the verbose `.cpp2`?

Are we trying to give space for a non-existent C2? Already C++ == C+1, and this is that + 1, so C2 is right I feel. :slightly_smiling_face: 

Anyhow if such a C2 language might exist in future (unlikely?), then it would require the `.h2` extension also, which we aren't trying to avoid.

Some naming conventions use `.hpp` against `.h` for C++ headers against C ones, but mostly `.h` is used. So nobody insists on avoiding `.h2` and using `.hpp2` for CPP2 I guess.

But if `.h2` is acceptable then the simplest would be `.c2` and not `.cpp2`. Note that it would be easier to distinguish from `.cpp` in file listings and would fit within the same (less) number of columns too.

**Comment by DyXel** [comment]
A quick search reveals that [C2 Programming Language](http://www.c2lang.org/) already exists and does in fact use the `.c2` extension.

> Anyhow if such a C2 language might exist in future (unlikely?), then it would require the .h2 extension also

AFAIK doing headers + translation unit is a dying tradition, modern languages try to implement modules which avoid most (all?) problems of having a preprocessor while being faster to compile. The aforementioned language implements modules, so it doesn't use headers and thus has no need for `.h2`.

Most C++ projects I have been involved in or read use `.hpp` ([Boost Header policy](https://www.boost.org/development/header.html) comes to mind a lot here), so actually, going full `.hpp2` makes more sense if we are talking about explicitly Cpp2 headers. In any case, from what I understood, headers are a transitional tool and when C++ modules are properly supported, that will be the main way of writing code.

**Comment by jamadagni** [comment]
Oh sorry for not checking first before I spoke. And yes, I hope we won't need `.h2` files in Cpp2 as well. Then `.cpp2` is fine if a bit verbose.

Maybe we should try `.cqq` [tongue_in_cheek] … ouch even [that is taken](https://github.com/ljt12138/Cqq-Project) (sort of… :smirk:)
